richtextfx/src/main/java/org/fxmisc/richtext/model/SegmentOps.java

```diff
@@ -0,0 +1,42 @@
+package org.fxmisc.richtext.model;
+
+import java.io.DataInputStream;
+import java.io.DataOutputStream;
+import java.io.IOException;
+
+
+/**
+ * Defines the operations which are supported on a specific segment type.
+ *
+ * @param <SEG>
+ * @param <S>
+ */
+public interface SegmentOps<SEG, S> {
+    public int length(SEG seg);
+
+    public char charAt(SEG seg, int index);
+
+    public String getText(SEG seg);
+
+    public SEG subSequence(SEG seg, int start, int end);
+
+    public SEG subSequence(SEG seg, int start);
+
+    public SEG append(SEG seg, String str);
```

Comment:
See [my PR](https://github.com/afester/RichTextFX/pull/5).

And funny enough, I did add `TextOps extends SegmentOps` 😄 . Although `ReadOnlyStyledDocument` still sees only `SegmentOps`. The only additional method is 

```java
public SEG create(String text, S style);
```

In `SegmentOps, the original methods `append` and `canJoin` are replaced by a single method

```java
public Optional<SEG> join(SEG currentSeg, SEG nextSeg);
```